vumi/transports/wechat/tests/test_wechat.py

```diff
@@ -23,23 +23,34 @@
 from vumi.persist.fake_redis import FakeRedis
 
 
-def request(transport, method, path='', params={}, data=None):
-    addr = transport.server.getHost()
-    token = transport.get_static_config().auth_token
+def request(transport, *a, **kw):
+    underrides = kw.setdefault('params', {})
 
     nonce = '1234'
     timestamp = '2014-01-01T00:00:00'
+    token = transport.get_static_config().auth_token
+
     good_signature = hashlib.sha1(
         ''.join(sorted([timestamp, nonce, token]))).hexdigest()
 
-    default_params = {
+    params = {
         'signature': good_signature,
         'timestamp': timestamp,
         'nonce': nonce,
     }
 
-    default_params.update(params)
-    path += '?%s' % (urlencode(default_params),)
+    params.update(underrides)
```

Comment:
The params handling is a bit odd. The `underrides` are possibly what is passed in by the user (i.e. `underrides = kw.setdefault('params', {})`) and they override the default params (i.e. `params.update(underrides)`).  Maybe it would be clearer to do:
```python
params = {
    'signature': good_signature,
    ...
}
params.update(kw.get('params', {}))
kw['params'] = params
```
?